packages/grid/src/Grid.tsx

```diff
@@ -1,81 +1,313 @@
+  );
+
+
+  const handleChangeMoving = useCallback(
+    (newColumns) => {
+      changeMappedColumns(newColumns);
+      onChangeColumns(newColumns);
+    },
+    [onChangeColumns],
+  );
 
   useEffect(() => {
-    themeRef.current = gridTheme<ITableTheme>(theme);
-  }, [theme]);
+    if (gridRef.current) gridRef?.current?.recomputeGridSize();
+    if (cache.current) cache.current.clearAll();
+  }, [mappedColumns, mappedRows]);
+
+  const {
+    header: { height: headerHeight = 0 },
+    totals: { height: totalsHeight = 0 },
+  } = themeRef.current;
 
   return (
-    <ContentWrapper theme={ themeRef.current }>
-      <HeadWrapper>
-        <TableHead style={ { transform: `translateX(-${tableScroll}px)` } } theme={ themeRef.current }>
-          { columns.map(({ title, width }: IColumn) => (
-            <HeaderStyled
-              width={ width }
-              key={ title }
-              theme={ themeRef.current.head }
-            >
-              { title }
-            </HeaderStyled>
-          )) }
-        </TableHead>
-      </HeadWrapper>
-      <TableWrapper
-        ref={ tableWrapperElement }
-        onScroll={ handleScroll }
-      >
-        <TableStyled>
-          { rows.map((row: IRowData) => (
-            <Row
-              row={ row }
-              key={ row.id }
-              columns={ columns }
+    <Wrapper theme={ themeRef.current } width={ width } changingColumns={ changingColumns }>
+      <HeaderWrapper
+        fullWidth={ fullWidth.current }
+        columns={ mappedColumns }
+        translateX={ scrollLeft }
+        onChangeWidth={ handleChangeWidth }
+        onChangeMoving={ handleChangeMoving }
+        changeChangingColumns={ changeChangingColumns }
+        theme={ themeRef.current }
+        shouldMovingColumns={ shouldMovingColumns }
+        shouldChangeColumnsWidth={ shouldChangeColumnsWidth }
+      />
+      <Body>
+        <VirtualisedGrid
+          ref={ gridRef as MutableRefObject<VirtualisedGrid> }
+          columnCount={ mappedColumns.length }
+          columnWidth={ ({ index }: any) => mappedColumns[index].width }
+          deferredMeasurementCache={ cache.current }
+          height={ height - +headerHeight - +totalsHeight }
+          cellRenderer={ cell }
+          rowCount={ mappedRows.length }
+          rowHeight={ cache.current.rowHeight }
+          width={ width - 2 }
+          onScroll={ handleScroll }
+        />
+      </Body>
+      { totals && (
+        <TotalBlock
+          width={ fullWidth.current }
+          translateX={ scrollLeft }
+          theme={ themeRef.current }
+        >
+          { mappedColumns.map((el: IColumn, index: number) => (
+            <TotalCell
               theme={ themeRef.current }
-              level={ 0 }
-              handleSelectRows={ handleSelectRows }
-              multiline={ multiline }
-            />
+              width={ mappedColumns.length === index + 1 ? el.width + 9 : el.width }
```

Comment:
Why you add 9 to to the last column?

Reply:
the last column should be wider by the width of the scroll